src/engine/TransitivePathBinSearch.h

```diff
@@ -54,6 +54,17 @@ struct BinSearchMap {
 
     return targetIds_.subspan(startIndex, range.size());
   }
+
+  // Retrieve pointer to equal id from `startIds_`, or nullptr if not present.
+  // This is used to get `Id`s that do do not depend on a specific `LocalVocab`,
+  // but instead are backed by the index.
```

Comment:
Why pointer and not `optional<Id>` ?

Reply:
Because where it's being called we need to return a span to a range of ids and optional does not support const references.